pkg/account/authorization.go

```diff
@@ -0,0 +1,185 @@
+	logger         log.Logger
+	configDBModule ConfigurationDBModule
+	next           Component
+}
+
+// MakeAuthorizationAccountComponentMW checks authorization and return an error if the action is not allowed.
+func MakeAuthorizationAccountComponentMW(logger log.Logger, configDBModule ConfigurationDBModule) func(Component) Component {
+	return func(next Component) Component {
+		return &authorizationComponentMW{
+			logger:         logger,
+			configDBModule: configDBModule,
+			next:           next,
+		}
+	}
+}
+
+// authorizationComponentMW implements Component.
+func (c *authorizationComponentMW) UpdatePassword(ctx context.Context, currentPassword, newPassword, confirmPassword string) error {
+	var action = UpdatePassword
+	var currentRealm = ctx.Value(cs.CtContextRealm).(string)
+
+	var config = dto.RealmConfiguration{}
+	var err error
+
+	if config, err = c.configDBModule.GetConfiguration(ctx, currentRealm); err != nil {
+		infos, _ := json.Marshal(map[string]string{
+			"currentRealm": currentRealm,
+		})
+		c.logger.Error("Error", "Configuration not found", "infos", string(infos))
+		return err
+	}
+
+	if !*config.APISelfPasswordChangeEnabled {
+		infos, _ := json.Marshal(map[string]string{
+			"Action":       action,
+			"currentRealm": currentRealm,
+		})
+		c.logger.Debug("ForbiddenError", "Password change disabled", "infos", string(infos))
+		return security.ForbiddenError{}
+	}
+
+	return c.next.UpdatePassword(ctx, currentPassword, newPassword, confirmPassword)
+}
+
+// authorizationComponentMW implements Component.
+func (c *authorizationComponentMW) GetCredentials(ctx context.Context) ([]api.CredentialRepresentation, error) {
+	// No restriction for this call
+	return c.next.GetCredentials(ctx)
+}
+
+func (c *authorizationComponentMW) GetCredentialRegistrators(ctx context.Context) ([]string, error) {
+	// No restriction for this call
+	return c.next.GetCredentialRegistrators(ctx)
+}
+
+func (c *authorizationComponentMW) UpdateLabelCredential(ctx context.Context, credentialID string, label string) error {
+	// No restriction for this call
+	return c.next.UpdateLabelCredential(ctx, credentialID, label)
+}
+
+func (c *authorizationComponentMW) MoveCredential(ctx context.Context, credentialID string, previousCredentialID string) error {
+	// No restriction for this call
+	return c.next.MoveCredential(ctx, credentialID, previousCredentialID)
+}
+
+func (c *authorizationComponentMW) DeleteCredential(ctx context.Context, credentialID string) error {
+	var action = DeleteCredential
+	var currentRealm = ctx.Value(cs.CtContextRealm).(string)
+
+	var config = dto.RealmConfiguration{}
+	var err error
+
+	if config, err = c.configDBModule.GetConfiguration(ctx, currentRealm); err != nil {
+		infos, _ := json.Marshal(map[string]string{
+			"currentRealm": currentRealm,
+		})
+		c.logger.Error("Error", "Configuration not found", "infos", string(infos))
+		return err
+	}
+
+	if !*config.APISelfAuthenticatorDeletionEnabled {
```

Comment:
Can't config.APISelfAuthenticatorDeletionEnabled be nil ?

Reply:
Good point, I have added a check